crates/core/types/receipt.rs

```diff
@@ -5,42 +5,52 @@ use crate::rlp::{
     structs::{Decoder, Encoder},
 };
 use bytes::Bytes;
-use ethereum_types::{Address, Bloom, H256};
+use ethereum_types::{Address, Bloom, BloomInput, H256};
 use serde::Serialize;
 
-use super::{BlockHash, BlockNumber, TxKind, TxType};
+use super::TxType;
 pub type Index = u64;
 
 /// Result of a transaction
 #[derive(Clone, Debug, PartialEq, Eq, Serialize)]
+#[serde(rename_all = "camelCase")]
+
 pub struct Receipt {
+    #[serde(rename = "type")]
     pub tx_type: TxType,
-    #[serde(with = "crate::serde_utils::bool")]
+    #[serde(with = "crate::serde_utils::bool", rename = "status")]
```

Comment:
I wonder if we should do all this rpc specific stuff inside the rpc crate (add a RpcReceipt type or something) and implement from/into to map to the "core" Receipt. Not in scope for this PR though. WDYT?

Reply:
Ah, I see there is one further down. Why does this one have the serde implementations?

Reply:
The `RpcReceipt` is a wrapper around this `Receipt` and some metadata:
```Rust
pub struct RpcReceipt {
    pub receipt: Receipt,
    pub tx_info: RpcReceiptTxInfo,
    pub block_info: RpcReceiptBlockInfo,
    ...
}
```
Should I unwrap the `Receipt` and store its fields directly?

Reply:
Maybe you could create a type like ReceiptInfo inside RPC that has the serde attributes. And then you can do a from/into the core Receipt. So that we have the Rpc stuff separated from the core stuff. Thoughts?

Reply:
Agreed!